app/verify_workflow.go

```diff
@@ -177,9 +182,18 @@ func getStepName(step string) string {
 }
 
 func getStepUses(step *actionlint.Step) *actionlint.String {
+	// TODO("naveen") :- Remove this check as this never happens
+	// errors during actionlint.Parse: [:16:3: "steps" section is
+	// missing in job "analysis" [syntax-check]]
+	// workflow-invalid-nil-steps.yml has nil steps
 	if step == nil || step.Exec == nil {
 		return nil
 	}
```

Comment:
`step.Exec == nil` can still happen right? I think you mean `step == nil` can never happen. If so, can remove that condition since we have a test for it already.

Reply:
Yes, I agree. I have fixed it.